### DISCUSSION TITLE
【重点】接下来要做的事

### DISCUSSION THREAD

**Discussion by Moonofweisheng** [Moonofweisheng/wot-design-uni#45] [Announcements]
**要增加的组件** 
目前计划中要新增的组件，朋友们有更多建议可以一起讨论，同时也欢迎有兴趣的同学认领开发。
- [x] WaterMark 水印
- [x] Swiper 轮播
- [x] Circle 环形进度条
- [x] Segmented 分段控制器
- [x] Tabbar 标签栏
- [x] Navbar 导航条
- [x] Table 表格
- [x] Sidebar 侧边栏
- [x] Fab 悬浮按钮
- [x] CountDown 倒计时
- [x] NumberKeyboard 数字键盘
- [x] PasswordInput 密码输入框
- [x] Form 表单
- [x] Textarea 文本域
- [ ] Signature 签名

**计划兼容的平台** 
在计划中的组件全部完成之后，我会开启兼容更多平台的工作。如果有更多平台的需求也可以提出。
- [x] 支付宝小程序
- [x] 钉钉小程序
- [ ] Vue2

**计划中的功能** 
- [x] 支持国际化
- [ ] 提供设计资源

**工程化改造** 
- [ ] 组件库工程改造为monorepo

**Comment by LTTGGj** [comment]
希望input及一些输入型组件可以增加formatter输入过滤或格式化

**Comment by LP-J** [comment]
加油ヾ(◍°∇°◍)ﾉﾞ

**Comment by liyuan1994** [comment]
希望增加列表和悬浮按钮组件

**Reply by Moonofweisheng** [reply]
可以简单描述一下期望的API和使用方式，或者给个参考。

**Reply by liyuan1994** [reply]
可以参考uniapp的uni-list和uni-fab

**Reply by Moonofweisheng** [reply]
OK，我会参考一下。

**Reply by Moonofweisheng** [reply]
fab悬浮按钮组件加入计划

**Reply by Moonofweisheng** [reply]
fab悬浮按钮组件已提供

**Comment by Moonofweisheng** [comment]
[v0.1.22](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.22)新增了 [Swiper 轮播图](https://wot-design-uni.cn/component/swiper.html)

**Comment by Moonofweisheng** [comment]
[v0.1.23](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.23)新增了 Segmented 分段器

**Comment by Moonofweisheng** [comment]
[v0.1.27](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.27)新增了 Tabbar 标签栏

**Comment by Moonofweisheng** [comment]
[v0.1.33](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.33)新增了 Navbar 导航栏

**Comment by Moonofweisheng** [comment]
[v0.1.39](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.39)新增了 Table 表格

**Comment by Moonofweisheng** [comment]
[v0.1.49](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.49)新增了 Sidebar 侧边栏

**Comment by Moonofweisheng** [comment]
[v0.1.57](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.1.57)新增了 Fab 悬浮动作按钮

**Comment by THUNDZENG** [comment]
1、希望 form 表单增加数据检查机制，类似与 uni-form 的 rules 检查，检查通过才允许 submit 。这是个硬需求，使用场景挺多的，急求！！
2、希望增加 省、市、区 地址联动选择组件。
@Moonofweisheng

**Reply by Moonofweisheng** [reply]
第一点开始开发  
第二点可以使用[col-picker](https://wot-design-uni.netlify.app/component/col-picker.html)

**Comment by Moonofweisheng** [comment]
[v0.2.0](https://github.com/Moonofweisheng/wot-design-uni/releases/tag/v0.2.0)新增了 Form 表单组件

**Comment by zhongwuhuang** [comment]
移动端单位别用px啊，移动端单位别用px啊，移动端单位别用px啊

**Reply by Moonofweisheng** [reply]
你可以使用[postcss-px-to-viewport](https://github.com/evrone/postcss-px-to-viewport)将组件库的px转成vw，和rpx一样都是百分比的。组件库作为通用库，需要考虑通用场景，暂时不会迁移至rpx，同样也可以使用其他的postcss插件将px转成rem或者其他单位。

**Reply by zhongwuhuang** [reply]
转肯定会出问题的，而且只能转样式部分的，动态的单位就转不了，uniapp推崇的就是rpx，个人理解手机端通用库就不应该用px啊，适配uniapp的其他几个下载量大的ui库都是用rpx的

**Reply by Moonofweisheng** [reply]
例如，当你要适配宽屏设备时，rpx无法适应的，这一点组件库应当提供通用的设计，而非百分比的单位，你提到的动态单位，我想应该是较少的，作为prop传入组件的大小，通常是支持字符串的，这意味着传入rpx也是被允许的，如果有不支持的组件，可以提出issue进行调整。

**Reply by zhongwuhuang** [reply]
适配宽屏设备rpx为啥不能适应？uniapp有个配置可以配置宽屏超过某个值时就以375为计算标准的，这个多好啊
"rpxCalcMaxDeviceWidth": 560, // rpx 计算所支持的最大设备宽度，单位 px，默认值为 960

**Reply by zhongwuhuang** [reply]
你说转单位我尝试过，有很多问题，后来就放弃了

**Reply by Moonofweisheng** [reply]
你知道rpxCalcMaxDeviceWidth，说明你也看了uni-app给的宽屏适配策略，对于无节制地使用rpx，实际上是并不被推荐的。关于单位的转换这是一个方案，等我空出时间会针对性的给出几个demo，目前来讲单位迁移到rpx是无计划的。

**Reply by MuyianKing** [reply]
支持px，uview我自己手动全部改成了px

**Comment by Moonofweisheng** [comment]
[2024开发计划](https://github.com/users/Moonofweisheng/projects/1)

**Comment by fengtang123** [comment]
在uniapp当中使用，本地的样式可以修改，打包后就不行

**Reply by Moonofweisheng** [reply]
给一下复现demo看看

**Comment by lirunsheng** [comment]
能否在toast组件中添加一些事件，比如加入轻提示开始时得事件或者消失时得事件！感谢开源组件wot-design-uni

**Reply by Moonofweisheng** [reply]
计划提供。

**Comment by kasnti** [comment]
索引列表会有计划支持吗

**Reply by Moonofweisheng** [reply]
会加入计划。

**Reply by kasnti** [reply]
感谢，期待

**Reply by kasnti** [reply]
看到出索引列表了，特来感谢，一直在关注

**Reply by Moonofweisheng** [reply]
感谢关注！

**Comment by kasnti** [comment]
能否支持抖音小程序？

**Reply by kasnti** [reply]
想问下博主是否有计划支持字节小程序，目前好像没有ui组件能支持它的，uni-ui实在是太坑了，里面隐藏的bug一堆，苦不堪言

**Reply by Moonofweisheng** [reply]
> 想问下博主是否有计划支持字节小程序，目前好像没有ui组件能支持它的，uni-ui实在是太坑了，里面隐藏的bug一堆，苦不堪言

会支持，可以参加issue里面关于字节小程序的讨论

**Comment by zhang-0-yu** [comment]
“wd-grid-item”组件上“icon”属性里面没有加上这个属性值“class-prefix”，只能用内部图标，切不能用外部图标。加上“class-prefix”属性就完美了

**Comment by elarbor** [comment]
作者好厉害啊👍

**Comment by elarbor** [comment]
其实咱们的组件库也不用内置图标，主要是太少不够用。使用[unocss](https://icones.js.org/)+[iconify](https://icones.js.org/)的图标库就很方便。二十多万的图标总有适合自己的
![image](https://github.com/user-attachments/assets/50b84ade-78e2-4685-ab40-4cd215f904c4)
![image](https://github.com/user-attachments/assets/d1b33108-90c4-4c6d-9112-4cc0546b5764)
![image](https://github.com/user-attachments/assets/f3555ff4-b1dc-4b3f-93cb-ed9ef8f56a0f)

希望咱们的组件库能直接内置

**Reply by Moonofweisheng** [reply]
确实[iconify](https://icones.js.org/)很好用，会考虑，感谢！

**Comment by wendell0316** [comment]
会支持头条小程序么

**Reply by Moonofweisheng** [reply]
还不支持，目前没有将兼容头条作为重点。

**Comment by waterbang** [comment]
请问 `signature`的需求是什么？ 可否直接使用 [Web_Crypto_API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API)

**Comment by zzl221zzl** [comment]
希望做一些内置样式，比如vant的https://vant.pro/vant/#/zh-CN/style

**Comment by QvQ12321y** [comment]
希望能做自定义状态栏和控制中心❤️

**Comment by yangli0107** [comment]
支不支持飞书小程序

**Comment by Moonofweisheng** [comment]
飞书小程序目前应该被飞书放弃了，你选择开发h5应用就行



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2025年10月14日(周二) 中午11:05
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Moonofweisheng/wot-design-uni] 【重点】接下来要做的事 (Discussion #45)




 
支不支持飞书小程序

**Comment by zhangqingnian** [comment]
wd-select-picker 组件的 搜索功能   搜索条件能不能对外暴露个方法 让开发者自己去定义筛选的条件
